File: Two Pointer/3 sum.py


def three_sum_value(nums,target):
    nums.sort()
    
    for i in range(len(nums)-2):
        low=i+1
        high=len(nums)-1
        while low < high:
            res = nums[i] + nums[low] + nums[high]
            if res == target:
                return True
            elif res < target:
                low+=1
            else:
                high-=1
    return False
            

File: Two Pointer/3 sum.go
package TwoPointer

import "sort"


func ThreeSumValue(nums []int,target int) bool {
	sort.Ints(nums)
	for i:=0; i<len(nums)-2;i++ {
		low := i+1
		high := len(nums)-1

		for low <  high {
			trible := nums[i] + nums[low] + nums[high]
			if target == trible{
			return true
			}else if target > trible {
				low++
			}else {
				high--
			}
		}
	}
	return false
}